src/commands/utils/clusters.ts

```diff
@@ -102,3 +110,71 @@ export async function getClusterProperties(
         return { succeeded: false, error: `Error invoking ${clusterName} managed cluster: ${ex}` };
     }
 }
+
+export async function determineClusterState(
+    target: AksClusterTreeItem,
+    clusterName: string
+): Promise<string | undefined> {
```

Comment:
My preference would be to return `Promise<errorable<string>>` and let callers decide whether to show the error message. This would make it easier to consume in other places (I'm thinking `startCluster` and `stopCluster`).